src/components/MAPI/organizations/OrganizationDetailGeneral/utils.ts

```diff
@@ -331,3 +333,80 @@ function mergeClusterSummaries(
     {}
   );
 }
+
+/**
+ * Get various statistics about the release versions of the given clusters.
+ * @param httpClientFactory
+ * @param auth
+ * @param clusters
+ */
+export async function fetchReleasesSummary(
+  httpClientFactory: HttpClientFactory,
+  auth: IOAuth2Provider,
+  clusters: capiv1alpha3.ICluster[]
+): Promise<ui.IOrganizationDetailReleasesSummary> {
+  const summary: ui.IOrganizationDetailReleasesSummary = {};
+
+  const releases = clusters
+    .map((c) => capiv1alpha3.getReleaseVersion(c))
+    .sort(compare);
+
+  summary.releasesInUseCount = releases.length;
+
+  if (summary.releasesInUseCount < 1) return Promise.resolve(summary);
+
+  summary.oldestReleaseVersion = releases[0];
+  summary.newestReleaseVersion = releases[releases.length - 1];
+
+  try {
+    const [
+      oldestReleasePromise,
+      newestReleasePromise,
+    ] = await Promise.allSettled([
```

Comment:
Using `Promise.allSettled()` instead of `Promise.all()`, so I could run the requests in parallel, but still be able to use each value if one of them decides to error out.